File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsjournal.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsJournal struct {
	CreatedOn          *time.Time                                   `json:"createdOn,omitempty"`
	HasChildren        *bool                                        `json:"hasChildren,omitempty"`
	ID                 *string                                      `json:"id,omitempty"`
	JournalCode        *string                                      `json:"journalCode,omitempty"`
	Metadata           *CodatDataContractsDatasetsMetadata          `json:"metadata,omitempty"`
	ModifiedDate       *time.Time                                   `json:"modifiedDate,omitempty"`
	Name               *string                                      `json:"name,omitempty"`
	ParentID           *string                                      `json:"parentId,omitempty"`
	SourceModifiedDate *time.Time                                   `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsJournalStatusEnum `json:"status,omitempty"`
	Type               *string                                      `json:"type,omitempty"`
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_journal.py
import dataclasses
import dateutil.parser
from ..shared import codat_datacontracts_datasets_journalstatus_enum as shared_codat_datacontracts_datasets_journalstatus_enum
from ..shared import codat_datacontracts_datasets_metadata as shared_codat_datacontracts_datasets_metadata
from dataclasses_json import dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsJournal:
    created_on: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdOn'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    has_children: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('hasChildren') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    journal_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('journalCode') }})
    metadata: Optional[shared_codat_datacontracts_datasets_metadata.CodatDataContractsDatasetsMetadata] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    parent_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parentId') }})
    source_modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceModifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    status: Optional[shared_codat_datacontracts_datasets_journalstatus_enum.CodatDataContractsDatasetsJournalStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    